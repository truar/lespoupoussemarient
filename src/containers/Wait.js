import React, { Component } from 'react';
import {
    Form, FormGroup, FormFeedback, Button, Input
  } from 'reactstrap';
import Section from '../components/Section';
import WrapperContent from '../components/WrapperContent';
import './Profile.css';

class Wait extends Component {

    render() {
        return (
            <Section id="Profile" className="Profile">
                <WrapperContent title="Les poupous se marient !">
                   <p>Le site sera disponible Ã  partir du 31 Janvier 2019.</p>
                </WrapperContent>
            </Section>
        );
    }

    renderAuthenticationForm() {
        const {error, guestCode, onChange, onSubmit} = this.props;
        return (
            <WrapperContent title="Les poupous se marient !">
                <Form onSubmit={onSubmit} id="login-form">
                    <FormGroup>
                        <Input invalid={error} onChange={onChange} type="text" name="guestCode" id="guestCode" placeholder="Entrez votre code..." value={guestCode}/>
                        <FormFeedback>Code incorrect</FormFeedback>
                    </FormGroup>
                    <Button>Se connecter</Button>
                </Form>
            </WrapperContent>
        );
    }

    renderLogged() {
        const dateDisplay = this.calculateRemainingDays();
        return (
            <WrapperContent title="Que l'aventure commence !">
                <p>{dateDisplay}</p>
            </WrapperContent>
        );
    }
}

export default Wait;