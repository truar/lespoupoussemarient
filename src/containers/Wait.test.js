import React from 'react';
import Wait from './Wait';
import { shallow } from 'enzyme';

it('renders without crashing', () => {
  const component = shallow(<Wait />);
  expect(component).toMatchSnapshot();
});

