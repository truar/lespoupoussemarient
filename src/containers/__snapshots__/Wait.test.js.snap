// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Wait />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <WrapperContent
        title="Les poupous se marient !"
      >
        <p>
          Le site sera disponible à partir du 31 Janvier 2019.
        </p>
      </WrapperContent>,
      "className": "Profile",
      "id": "Profile",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <p>
          Le site sera disponible à partir du 31 Janvier 2019.
        </p>,
        "title": "Les poupous se marient !",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Le site sera disponible à partir du 31 Janvier 2019.",
        },
        "ref": null,
        "rendered": "Le site sera disponible à partir du 31 Janvier 2019.",
        "type": "p",
      },
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <WrapperContent
          title="Les poupous se marient !"
        >
          <p>
            Le site sera disponible à partir du 31 Janvier 2019.
          </p>
        </WrapperContent>,
        "className": "Profile",
        "id": "Profile",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <p>
            Le site sera disponible à partir du 31 Janvier 2019.
          </p>,
          "title": "Les poupous se marient !",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Le site sera disponible à partir du 31 Janvier 2019.",
          },
          "ref": null,
          "rendered": "Le site sera disponible à partir du 31 Janvier 2019.",
          "type": "p",
        },
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
